PatternPal/PatternPal.Tests/Checks/RelationCheckTests.cs

```diff
@@ -0,0 +1,335 @@
+
+        //check for the UsedNode inheriting from the UsesNode, a relation which does not exist
+        ClassCheck usedClass = Class(
+            Priority.Low,
+            Inherits(
+                Priority.Low,
+                usesClass.Result));
+
+        ICheckResult result = usedClass.Check(
+            ctx,
+            usedNode);
+
+        return Verifier.Verify(result);
+    }
+
+    [Test]
+    public Task Implements_Check_Returns_Correct_True_Result()
+    {
+        //a syntax graph with an inheritance relation from Child to Parent
+        SyntaxGraph graph = EntityNodeUtils.CreateImplementationRelation();
+        RecognizerContext ctx = new()
+        {
+            Graph = graph
+        };
+
+        //the Child node from the syntax graph 
+        INode childInterfaceNode = graph.GetAll()["Child"];
+        //the Parent node from the syntax graph 
+        INode parentInterfaceNode = graph.GetAll()["Parent"];
+
+        //check for the interface
+        InterfaceCheck parentInterfaceCheck = Interface(Priority.Low);
+        //checking the check to set MatchedEntity to interfaceNode
+        parentInterfaceCheck.Check(
+            ctx,
+            parentInterfaceNode);
+
+        //check for the Child node implementing the interface node
+        InterfaceCheck childInterfaceCheck = Interface(
+            Priority.Low,
+            Implements(
+                Priority.Low,
+                parentInterfaceCheck.Result));
+
+        ICheckResult result = childInterfaceCheck.Check(
+            ctx,
+            childInterfaceNode);
+
+        return Verifier.Verify(result);
+    }
+
+    [Test]
+    public Task Implements_Check_Returns_Correct_False_Result()
+    {
+        //a syntax graph with an inheritance relation from Child to Parent
+        SyntaxGraph graph = EntityNodeUtils.CreateImplementationRelation();
+        RecognizerContext ctx = new()
+        {
+            Graph = graph
+        };
+
+        //the Child node from the syntax graph 
+        INode childInterfaceNode = graph.GetAll()["Child"];
+        //the Parent node from the syntax graph 
+        INode parentInterfaceNode = graph.GetAll()["Parent"];
+
+        //check for the child interface
+        InterfaceCheck parentInterfaceCheck = Interface(Priority.Low);
+        //checking the check to set MatchedEntity to interfaceNode
+        parentInterfaceCheck.Check(
+            ctx,
+            childInterfaceNode);
+
+        //check for the parent node implementing the interface node, which isnt the case
+        InterfaceCheck childInterfaceCheck = Interface(
+            Priority.Low,
+            Implements(
+                Priority.Low,
+                parentInterfaceCheck.Result));
+
+        ICheckResult result = childInterfaceCheck.Check(
+            ctx,
+            parentInterfaceNode);
+
+        return Verifier.Verify(result);
+    }
+
+    [Test]
+    public Task Creates_Check_Returns_Correct_True_Result()
```

Comment:
Zou je in de comments deze test iets duidelijker kunnen zijn in het verschil tussen creates en uses. Want nu zie ik de term uses erg vaak terug komen waardoor ik me ga afvragen of creates wel wordt getest of dat uses wordt getest. En nu snap ik dat het verschil niet zo groot is maar daar kan denk ik in de comments iets meer duidelijkheid over komen. Dit kan ook doormiddel van een grotere comment boven de test. Het is mij nu niet helemaal duidelijk waar deze anders is dan de Creates_Check_Returns_Correct_False_Result(). Dus probeer dat ook te verduidelijken in een comment